Details basic auth VS BO (#2224)
ibexa/documentation-developer: docs/api/rest_api/rest_api_authentication.md

    Back Office uses the REST API too (for some parts like the Location tree or the Calendar) on its own domain.
    
    * If the Back Office SiteAccess matches `//admin.example.com` (through `Map\Host`, `HostElement` or `HostText`), it will call the REST API under `//admin.example.com/api/ibexa/v2`;
    * If the Back Office SiteAccess matches `//localhost/admin` (through `URIElement`, `Map\URI` or `Regex\URI`), it will call the REST API under `//localhost/api/ibexa/v2` because SiteAccess matching with REST isn't enabled at URL level.

>     If you set up basic authentification for `pattern: ^/api/ibexa/v2` to have it on your front office on both production and developments environments, your development environment's Back Office doesn't work properly anymore.

Reviewer: ```suggestion
    If you enable basic authentication for pattern: ^/api/ibexa/v2 to use it in your front office across both production and development environments, your development environment's Back Office won't work correctly.
```

>     This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification. Some 

Author: ```suggestion
    This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification.
```

>     This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office  and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.

Author: ```suggestion
    This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```

>     This Back Office will try to access REST through the same URL than front office. Even if you're logged in your Back Office and the [X-SiteAccess header](rest_requests.md#siteaccess) is used, the firewall denies access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.

Reviewer: ```suggestion
    This Back Office tries to access REST through the same URL as the front office. Even when logged in Back Office  and using the [X-SiteAccess header](rest_requests.md#siteaccess), the firewall blocks access to REST as you're not logged through basic authentification. Therefore, some Back Office features don't work.
```
